### PR TITLE
Fix api docs in RNN, Transformer, layer_norm, WeightNormParamAttr

### PR DESCRIPTION
**Pull request by guoshengCS**
### PR types

Others

### PR changes

Docs

### Describe

Fix api docs in SimpleRNN, LSTM, GRU, Transformer, layer_norm, label_smooth, crf_decoding, WeightNormParamAttr.
Remove argument `name` in SimpleRNN, LSTM, GRU since it is not used.

### CODE REVIEW

**Comment by guoshengCS** [comment]
image
image
image
image
image
image
image
image

附中文文档：https://github.com/PaddlePaddle/FluidDoc/pull/2975

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -845,49 +845,48 @@ def crf_decoding(input, param_attr, label=None, length=None):
     ${comment}
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/nn.py:846]: 这里应该需要加个空行 否则会有System Warning

File: python/paddle/nn/layer/rnn.py [hunk 2]
```diff
@@ -1252,8 +1249,7 @@ def __init__(self,
                  weight_ih_attr=None,
                  weight_hh_attr=None,
                  bias_ih_attr=None,
-                 bias_hh_attr=None,
-                 name=None):
+                 bias_hh_attr=None):
```
Comment (TCChenlong) [inline python/paddle/nn/layer/rnn.py]: 1210 行的 Attributes: 需要换成Variables: 不然解析会出问题

File: python/paddle/nn/layer/rnn.py [hunk 3]
```diff
@@ -1464,7 +1454,6 @@ class GRU(RNNBase):
             index are not less than the valid length are treated as paddings.
 
     Returns:
-        (outputs, final_states)
```
Comment (TCChenlong) [inline python/paddle/nn/layer/rnn.py]: 1469 同上

File: python/paddle/nn/layer/transformer.py [hunk 4]
```diff
@@ -643,7 +643,7 @@ class TransformerDecoderLayer(Layer):
             for linear in FFN. Otherwise, the three sub-layers all uses it as
             `weight_attr` to create parameters. Default: None, which means the
             default weight parameter property is used. See usage for details
-            in :ref:`api_fluid_ParamAttr` . 
+            in :ref:`api_guide_ParamAttr` .
```
Comment (TCChenlong) [inline python/paddle/nn/layer/transformer.py]: api_guide_ParamAttr  - > api_paddle_fluid_param_attr_ParamAttr

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -845,49 +845,48 @@ def crf_decoding(input, param_attr, label=None, length=None):
...
 
     Args:
-        input(${emission_type}): ${emission_comment}
+        input(Tensor): ${emission_comment}
 
         param_attr (ParamAttr|None): To specify the weight parameter attribute.
             Default: None, which means the default weight parameter property is
-            used. See usage for details in :ref:`api_fluid_ParamAttr` .
+            used. See usage for details in :ref:`api_guide_ParamAttr` .
```
Comment (TCChenlong) [inline python/paddle/fluid/layers/nn.py]: api_guide_ParamAttr -> api_paddle_fluid_param_attr_ParamAttr

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (guoshengCS) [reply python/paddle/fluid/layers/nn.py:846]: Done. Thanks

File: python/paddle/nn/layer/rnn.py (hunk 3 above)
Comment (guoshengCS) [reply python/paddle/nn/layer/rnn.py]: Done. Thanks

File: python/paddle/nn/layer/transformer.py (hunk 4 above)
Comment (guoshengCS) [reply python/paddle/nn/layer/transformer.py]: Done. Thanks

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (guoshengCS) [reply python/paddle/fluid/layers/nn.py]: Done. Thanks

**Review by TCChenlong — DISMISSED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]
LGTM for core.ops

**Pull request merged**